src/app/services/xml-parsers/structure-xml-parser.service.ts

```diff
@@ -21,63 +21,129 @@ export class StructureXmlParserService {
-        pages[page.id] = page;
-        pagesIndexes.push(page.id);
+        this.parseDocument(document, pages, pagesIndexes);
       }
-      console.log(pages);
     }
 
     return {
       pages,
       pagesIndexes,
     };
   }
+
+  parseDocumentPages(pageElements: NodeListOf<Element>, pagesNumber: number, pages: Map<Page>, pagesIndexes: string[]) {
+    pageElements.forEach((element, i) => {
+      let pageContent: XMLElement[] = [];
+
+      if (pagesNumber === 1) {
+        pageContent = getElementsAfterTreeNode(element);
+      }
+
+      if (pagesNumber !== 1) {
+        const isFirstPage = i === 0;
+        const isLastPage = i === pagesNumber - 1;
+
+        if (isFirstPage) {
+          pageContent = getElementsBetweenTreeNode(element, pageElements[i + 1]);
+        }
+
+        if (isLastPage) {
+          pageContent = getElementsAfterTreeNode(element);
+        }
+
+        if (!isFirstPage && !isLastPage) {
+          pageContent = getElementsBetweenTreeNode(element, pageElements[i + 1]);
+        }
+      }
+
+      pageContent = this.removeBackNodes(pageContent);
+
+      const page: Page = {
+        id: element.getAttribute('xml:id') || 'page_' + (pagesIndexes.length + 1),
+        label: element.getAttribute('n') || 'Page ' + (pagesIndexes.length + 1),
+        originalContent: pageContent,
+        parsedContent: this.parsePageContent(pageContent),
+      };
+      pages[page.id] = page;
+      pagesIndexes.push(page.id);
+    });
+  }
+
+  parseDocument(document: XMLElement, pages: Map<Page>, pagesIndexes: string[]) {
+    const mainText = document.querySelector('text');
+    let content = Array.from(mainText.childNodes);
+    const contentFront = content.filter((c) => c.nodeName === 'front');
+    const hasFrontOriginalContent = this.hasFrontOriginalContent(contentFront[0] as HTMLElement);
+    content = this.removeBackNodes(content as XMLElement[]);
+
+    if (hasFrontOriginalContent) {
+      const frontPage: Page = {
+        id: 'page_front',
+        label: 'Front',
+        originalContent: contentFront as XMLElement[],
+        parsedContent: this.parsePageContent(contentFront as OriginalEncodingNodeType[]).filter(c => !!c.content),
+      };
+      pages[frontPage.id] = frontPage;
+      pagesIndexes.push(frontPage.id);
+      content = content.filter((c) => c.nodeName !== 'front');
+    }
+
+    const page: Page = {
+      id: 'page_1',
+      label: 'Main Text',
+      originalContent: content as XMLElement[],
+      parsedContent: this.parsePageContent(content as OriginalEncodingNodeType[]).filter(c => !!c.content),
+    };
+
+    pages[page.id] = page;
+    pagesIndexes.push(page.id);
+  }
+
+  parsePageContent(pageContent: OriginalEncodingNodeType[]) {
+   const parsedContent = [];
+   pageContent.map((child: XMLElement) => {
+      if (isNestedInElem(child, 'front') || child.nodeName === 'front') {
+        if (child.nodeType === 3 || isNestedInElem(child, '', [{ key: 'type', value:'document_front' }])) {
```

Comment:
I would put this value in a separate property of the parser, so that it can eventually be changed if needed.

Reply:
Could the property name be `frontOriginalContentAttr` or what?